File: backend/src/main/kotlin/com/localbridge/backend/entity/Evidence.kt
package com.localbridge.backend.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "evidences")
data class Evidence(
    @Id
    @Column(length = 36)
    val id: String,

    @Column(name = "result_id", nullable = false, length = 36)
    val resultId: String,

    @Column(nullable = false, length = 50)
    @Enumerated(EnumType.STRING)
    val type: EvidenceType,

    @Column(name = "file_path", nullable = false, length = 500)
    val filePath: String,

    @Column(name = "mime_type", nullable = false, length = 100)
    val mimeType: String,

    @Column(name = "file_size")
    val fileSize: Long? = null,

    @Column(name = "thumbnail_path", length = 500)
    val thumbnailPath: String? = null,

    @Column(name = "s3_key", length = 500)
    val s3Key: String? = null, // S3にアップロード後のキー

    @Column(name = "created_at", nullable = false)
    val createdAt: Instant = Instant.now()
)

enum class EvidenceType {
    IMAGE,
    VIDEO
}

File: frontend/src/domain/entities/Evidence.ts
export class Evidence {
  readonly id: string
  readonly resultId: string
  readonly type: 'image' | 'video'
  readonly filePath: string // OPFSでのファイルパス
  readonly mimeType: string
  readonly createdAt: string // ISO 8601 UTC形式（ローカルで発行）
  readonly fileSize?: number // ファイルサイズ(bytes)
  readonly thumbnailPath?: string // サムネイルのパス(オプション)

  constructor(
    id: string,
    resultId: string,
    type: 'image' | 'video',
    filePath: string,
    mimeType: string,
    createdAt: string,
    fileSize?: number,
    thumbnailPath?: string
  ) {
    this.id = id
    this.resultId = resultId
    this.type = type
    this.filePath = filePath
    this.mimeType = mimeType
    this.createdAt = createdAt
    this.fileSize = fileSize
    this.thumbnailPath = thumbnailPath
  }
}
